### ISSUE TITLE
Setting initial duration throws a NullPointerException

### ISSUE THREAD

**Issue by amugofjava** [ryanheise/just_audio#283]
**Which API doesn't behave as documented, and how does it misbehave?**
Attempting to set an initial duration either via:
```
      await _player.setUrl(
               'https://s3.amazonaws.com/scifri-segments/scifri201711241.mp3',
               initialPosition: Duration(seconds: 10));
```
For example or:
```
      await _player.setAudioSource(
        AudioSource.uri(
          Uri.parse(
              "https://s3.amazonaws.com/scifri-segments/scifri201711241.mp3"),
          tag: AudioMetadata(
            album: "Science Friday",
            title: "From Cat Rheology To Operatic Incompetence",
            artwork:
                "https://media.wnyc.org/i/1400/1400/l/80/1/ScienceFriday_WNYCStudios_1400.jpg",
          ),
        ),
        initialPosition: Duration(seconds: 1),
      );
```
Issue occurs for both local and streamed files.

**Minimal reproduction project**

Please find a version of your example application that reproduces this error [here](https://github.com/amugofjava/just_audio)

**To Reproduce (i.e. user steps, not code)**

Run the example and it should fail straight away

**Error messages**

```
W/System.err(32373): java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.Integer.intValue()' on a null object reference
W/System.err(32373): 	at com.ryanheise.just_audio.AudioPlayer.onTimelineChanged(AudioPlayer.java:191)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl$PlaybackInfoUpdate.lambda$run$0$ExoPlayerImpl$PlaybackInfoUpdate(ExoPlayerImpl.java:1410)
W/System.err(32373): 	at com.google.android.exoplayer2.-$$Lambda$ExoPlayerImpl$PlaybackInfoUpdate$N_S5kRfhaRTAkH28P5luFgKnFjQ.invokeListener(Unknown Source:2)
W/System.err(32373): 	at com.google.android.exoplayer2.BasePlayer$ListenerHolder.invoke(BasePlayer.java:279)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.invokeAll(ExoPlayerImpl.java:1496)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.access$100(ExoPlayerImpl.java:56)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl$PlaybackInfoUpdate.run(ExoPlayerImpl.java:1408)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.notifyListeners(ExoPlayerImpl.java:1326)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.updatePlaybackInfo(ExoPlayerImpl.java:954)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSourcesInternal(ExoPlayerImpl.java:1062)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSources(ExoPlayerImpl.java:384)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSources(ExoPlayerImpl.java:379)
W/System.err(32373): 	at com.google.android.exoplayer2.ExoPlayerImpl.setMediaSource(ExoPlayerImpl.java:363)
W/System.err(32373): 	at com.google.android.exoplayer2.SimpleExoPlayer.setMediaSource(SimpleExoPlayer.java:1499)
W/System.err(32373): 	at com.ryanheise.just_audio.AudioPlayer.load(AudioPlayer.java:525)
W/System.err(32373): 	at com.ryanheise.just_audio.AudioPlayer.onMethodCall(AudioPlayer.java:289)
W/System.err(32373): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:233)
W/System.err(32373): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:85)
W/System.err(32373): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:692)
W/System.err(32373): 	at android.os.MessageQueue.nativePollOnce(Native Method)
W/System.err(32373): 	at android.os.MessageQueue.next(MessageQueue.java:335)
W/System.err(32373): 	at android.os.Looper.loop(Looper.java:183)
W/System.err(32373): 	at android.app.ActivityThread.main(ActivityThread.java:7656)
W/System.err(32373): 	at java.lang.reflect.Method.invoke(Native Method)
W/System.err(32373): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
W/System.err(32373): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
I/flutter (32373): #0      AudioPlayer._load (package:just_audio/just_audio.dart:637:11)
```

**Expected behavior**

Playback should start from the `Duration` value. If you set `preload` to false the exception is not thrown, but you can no longer hear any audio playback; if I set `preload` to false and manually call the `.load()` method the above exception is thrown. If I remove the Duration playback works correctly.

**Smartphone (please complete the following information):**
 - Device: Android 11 emulator.
 - OS: Android

**Flutter SDK version**
```
[✓] Flutter (Channel stable, 1.22.5, on Linux, locale en_GB.UTF-8)
    • Flutter version 1.22.5 at /home/ben/Development/flutter
    • Framework revision 7891006299 (5 weeks ago), 2020-12-10 11:54:40 -0800
    • Engine revision ae90085a84
    • Dart version 2.10.4

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /media/ubuntudata/Development/android/sdk
    • Platform android-30, build-tools 30.0.3
    • Java binary at: /home/ben/Development/android/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Android Studio (version 4.0)
    • Android Studio at /home/ben/Development/android/android-studio
    • Flutter plugin version 50.0.1
    • Dart plugin version 193.7547
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] VS Code (version 1.52.1)
    • VS Code at /usr/share/code
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64 • Android 11 (API 30) (emulator)

• No issues found!
```

**Additional context**
I used to use the `seek` method after setting the URL to set the start position, but I don't think I can do this any more as you can't do it before a file is playing. Do I have that correct?

Many thanks for a great library.

**Comment by ryanheise** [comment]
Thank you for the bug report, I will investigate this today.

**Comment by ryanheise** [comment]
I have just fixed the bug on git master which you can try with this pubspec dependency:

```dart
just_audio: 
    git:
      url: git://github.com/ryanheise/just_audio.git
      path: just_audio
```

Let me know how it goes for you and if it functions correctly, I'll include this in the next release.

**Comment by amugofjava** [comment]
Yes, that works perfectly now. Thank you for fixing it so quickly.

**Comment by ryanheise** [comment]
Great! Thanks for letting me know it's working now.

**Comment by ryanheise** [comment]
This fix is now available in release 0.6.6.
